Samples/MediaFoundationTransformDecoder/cpp/CHWMFT.cpp

```diff
@@ -489,15 +489,46 @@ HRESULT CHWMFT::OnDrain(
     const UINT32 un32StreamID)
 {
     HRESULT hr = S_OK;
-
     TraceString(CHMFTTracing::TRACE_INFORMATION, L"%S(): Enter", __FUNCTION__);
-
     do
     {
         CAutoLock lock(&m_csLock);
-                
-        m_dwStatus |= (MYMFT_STATUS_DRAINING);
-
+        if (m_pOutputSampleQueue->IsQueueEmpty())
+        {
+            IMFMediaEvent* pDrainCompleteEvent = NULL;
+            hr = MFCreateMediaEvent(METransformDrainComplete, GUID_NULL, S_OK, NULL, &pDrainCompleteEvent);
+            if (FAILED(hr))
+            {
+                break;
+            }
+            /*******************************
+            ** Todo: This MFT only has one
+            ** input stream, so the drain
+            ** is always on stream zero.
+            ** Update this is your MFT
+            ** has more than one stream
+            *******************************/
+            hr = pDrainCompleteEvent->SetUINT32(MF_EVENT_MFT_INPUT_STREAM_ID, 0);
+            if (FAILED(hr))
+            {
+                break;
```

Comment:
There should also be another `SAFERELEASE(pDrainCompleteEvent);` if this `SetUINT32` method fails to prevent a memory leak